### PR TITLE
feat(llm): 🏷️ add memo/tag input in some coins send flow

### PR DESCRIPTION
**Pull request by thesan**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - ...

### 📝 Description

https://github.com/user-attachments/assets/88380de6-9045-4590-b954-ff83177ac44d

Add a memo tag input on all coins where it had been implemented on the summary step except Stellar. Stellar will be done on a follow up PR.

List of the coins:
- Algorand
- Cardano
- Casper
- Cosmos
- Crypto_org
- Hedera
- Internet_computer
- Solana
- Stacks
- Stellar
- TON
- XRP

### ❓ Context

- [LIVE-9634]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-9634]: https://ledgerhq.atlassian.net/browse/LIVE-9634?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/families/algorand/MemoTagInput.tsx
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/families/algorand/MemoTagInput.tsx:1]: Is it possible to make one generic MemoTag? like 

```
import React from "react";

import { AnimatedInput } from "@ledgerhq/native-ui";
import { MemoTagInputProps } from "LLM/features/MemoTag/types";

// Make the component generic, accepting a transaction type `T`.
export default function MemoTagInput<T>({ onChange, ...inputProps }: MemoTagInputProps<T>) {
  const [value, setValue] = React.useState("");

  const handleChange = (text: string) => {
    const value = text;
    setValue(value);
    onChange({
      patch: { memo: value || undefined },
      isEmpty: !value,
    });
  };

  return <AnimatedInput {...inputProps} value={value} onChangeText={handleChange} />;
}

```

and then use it in each family ? 
```
import type { Transaction as CardanoTransaction } from "@ledgerhq/live-common/families/cardano/types";

<MemoTagInput<CardanoTransaction>
  onChange={handleCardanoChange}
  /* other props */
/>;
```

File: apps/ledger-live-mobile/src/screens/SendFunds/02-SelectRecipient.tsx [hunk 1]
```diff
@@ -146,7 +159,17 @@ export default function SendSelectRecipient({ navigation, route }: Props) {
     setTransaction(bridge.updateTransaction(transaction, {}));
   }, [setTransaction, account, parentAccount, transaction]);
 
-  const onPressContinue = useCallback(async () => {
+  const [memoTagDrawerState, setMemoTagDrawerState] = useState<"INITIAL" | "SHOWING" | "SHOWN">(
+    "INITIAL",
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/SendFunds/02-SelectRecipient.tsx:163]: maybe use a dedicated enum or type?

File: apps/ledger-live-mobile/src/families/algorand/MemoTagInput.tsx
Comment (thesan) [reply apps/ledger-live-mobile/src/families/algorand/MemoTagInput.tsx:1]: Nice one 🙌 
It really reduced the boilerplate on every coins

**Review by mcayuelas-ledger — APPROVED** [review]
LGTM

**Review by Wozacosta — APPROVED** [review]

**Pull request merged**